### ISSUE TITLE
Vanishing validation accuracy

### ISSUE THREAD

**Issue by ronykalfarisi** [illidanlab/urgent-care-comparative#7]
Hi @af1tang ,
I successfully run the preprocess.py script to get the necessary input data. When I run the main.py, using the following arguments

```
================================================================================
                                      Opts                                      
--------------------------------------------------------------------------------
                           features_dir: local_mimic/save/X19.npy               
                          auxiliary_dir: local_mimic/save/w2v.npy               
                                  y_dir: local_mimic/save/y                     
                                  model: lstm                                   
                                   task: mort                                   
                            hidden_size: 256                                    
                                nepochs: 100                                    
                             batch_size: 32                                     
                          learning_rate: 0.005                                  
                         checkpoint_dir: local_mimic/save/checkpoint            
================================================================================

```

 I got the following result.

```
KFold #1
Train on 4211 samples, validate on 1053 samples
Epoch 1/1
 - 20s - loss: 0.4878 - acc: 0.7927 - val_loss: 0.5706 - val_acc: 0.7559

Epoch 00001: val_loss improved from -inf to 0.57057, saving model to local_mimic/save/checkpoint/lstm-mort-x19-w2v/best_model
Train on 4211 samples, validate on 1053 samples
Epoch 1/1
 - 9s - loss: 0.4065 - acc: 0.8195 - val_loss: 0.8074 - val_acc: 0.5926

Epoch 00001: val_loss improved from 0.57057 to 0.80737, saving model to local_mimic/save/checkpoint/lstm-mort-x19-w2v/best_model
Train on 4211 samples, validate on 1053 samples
Epoch 1/1
 - 9s - loss: 5.1078 - acc: 0.6350 - val_loss: 16.1181 - val_acc: 0.0000e+00

Epoch 00001: val_loss improved from 0.80737 to 16.11810, saving model to local_mimic/save/checkpoint/lstm-mort-x19-w2v/best_model
Train on 4211 samples, validate on 1053 samples
Epoch 1/1
 - 9s - loss: 7.3188 - acc: 0.5438 - val_loss: 16.1181 - val_acc: 0.0000e+00

Epoch 00001: val_loss did not improve from 16.11810
Train on 4211 samples, validate on 1053 samples
Epoch 1/1
 - 9s - loss: 7.3720 - acc: 0.5405 - val_loss: 16.1181 - val_acc: 0.0000e+00

Epoch 00001: val_loss did not improve from 16.11810
Train on 4211 samples, validate on 1053 samples
Epoch 1/1
 - 9s - loss: 7.3830 - acc: 0.5398 - val_loss: 16.1181 - val_acc: 0.0000e+00

Epoch 00001: val_loss did not improve from 16.11810
Train on 4211 samples, validate on 1053 samples
Epoch 1/1

...
```

The Epoch number is not changing. Also, I notice the validation accuracy is vanishing and  validation loss is not improving. Is it supposed to be like that? Thanks

**Comment by ronykalfarisi** [comment]
Also, I found the following for KFold #2 

```
KFold #2
Train on 4212 samples, validate on 1054 samples
Epoch 1/1
 - 10s - loss: 0.5977 - acc: 0.7870 - val_loss: 0.8320 - val_acc: 0.5797

Epoch 00001: val_loss improved from -inf to 0.83203, saving model to local_mimic/save/checkpoint/lstm-mort-x19-w2v/best_model
Train on 4212 samples, validate on 1054 samples
Epoch 1/1
 - 9s - loss: 0.4154 - acc: 0.8127 - val_loss: 0.8530 - val_acc: 0.5655

Epoch 00001: val_loss improved from 0.83203 to 0.85299, saving model to local_mimic/save/checkpoint/lstm-mort-x19-w2v/best_model
Train on 4212 samples, validate on 1054 samples
Epoch 1/1
 - 9s - loss: 0.3919 - acc: 0.8208 - val_loss: 0.8096 - val_acc: 0.5778

Epoch 00001: val_loss did not improve from 0.85299
Train on 4212 samples, validate on 1054 samples
Epoch 1/1
 - 9s - loss: 7.8134 - acc: 0.4786 - val_loss: 1.1921e-07 - val_acc: 1.0000

Epoch 00001: val_loss did not improve from 0.85299
Train on 4212 samples, validate on 1054 samples
Epoch 1/1
 - 9s - loss: 9.9659 - acc: 0.3749 - val_loss: 1.1921e-07 - val_acc: 1.0000

Epoch 00001: val_loss did not improve from 0.85299
Train on 4212 samples, validate on 1054 samples
Epoch 1/1
 - 9s - loss: 9.9698 - acc: 0.3746 - val_loss: 1.1921e-07 - val_acc: 1.0000

Epoch 00001: val_loss did not improve from 0.85299
Train on 4212 samples, validate on 1054 samples
Epoch 1/1
 - 9s - loss: 9.9545 - acc: 0.3756 - val_loss: 1.1921e-07 - val_acc: 1.0000

...

```
